### PR TITLE
feat(drivers): add fine-grained control for link signing

### PR DESCRIPTION
**Pull request by WhiteElytra**
Provide link signature authentication option for Local Driver.
Reuse the original Sign function for link signature.
Clear the sign parameter for URLs that don't require a signature.

### CODE REVIEW

**Comment by WhiteElytra** [comment]
After testing, only proxy url works, I will push a new commit to support direct url. (Tonight)

**Comment by Xhofe** [comment]
I can't understand why you do this?

**Comment by WhiteElytra** [comment]
ok i'll try it tonight.

**Comment by WhiteElytra** [comment]
It's done! Tested on Local Driver and Aliyundrive Driver.

File: public/dist/README.md [hunk 1]
```diff
@@ -1 +0,0 @@
-## Put dist of frontend here.
```
Comment (Xhofe) [inline public/dist/README.md:1]: Do not delete this file please, otherwise it will cause a ci failure.

File: server/handles/fsread.go [hunk 2]
```diff
@@ -260,16 +265,23 @@ func FsGet(c *gin.Context) {
...
 		}
 		if storage.Config().MustProxy() || storage.GetStorage().WebProxy {
+			query := "?"
+			if needSign(meta, reqPath) {
+				query = query + "sign=" + sign.Sign(reqPath)
+			}
+			if query == "?" {
+				query = ""
+			}
 			if storage.GetStorage().DownProxyUrl != "" {
```
Comment (Xhofe) [inline server/handles/fsread.go:275]: For `DownProxyUrl`, sign param is required regardless of whether sign is turned on in the settings due to its implementation.

File: server/handles/fsread.go [hunk 3]
```diff
@@ -362,3 +374,19 @@ func FsOther(c *gin.Context) {
 	}
 	common.SuccessResp(c, res)
 }
+
+func needSign(meta *model.Meta, path string) bool {
```
Comment (Xhofe) [inline server/handles/fsread.go:378]: This function is duplicated with `isEncrypt`, consider merging

**Review by Xhofe — CHANGES_REQUESTED** [review]

**Review by Xhofe — APPROVED** [review]

**Pull request merged**